File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    TreeNode* recursive(TreeNode* node, int val){
        if(node == nullptr){
            return new TreeNode(val);
        }
        if(val < node->val){
            node-> left = recursive(node->left, val);
        }else if(val > node->val){
            node-> right = recursive(node->right, val);
        }

        return node;
    }
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        return recursive(root, val);
    }
};

File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def recursive(self, node, val):
        if node is None:
            return TreeNode(val)
        
        if val < node.val:
            node.left = self.recursive(node.left, val)
        elif val > node.val:
            node.right = self.recursive(node.right, val)
        
        return node

    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        return self.recursive(root, val)
        